mesa/visualization/templates/js/runcontrol.js

```diff
@@ -43,7 +43,15 @@ var fpsControl = $('#fps').slider({
 var sidebar = $("#sidebar");
 
 // WebSocket Stuff
-var ws = new WebSocket("ws://127.0.0.1:" + port + "/ws"); // Open the websocket connection
+// Support TLS-specific URLs, when appropriate
+if (window.location.protocol == "https:") {
+    var ws_scheme = "wss://";
+} else {
+    var ws_scheme = "ws://"
+};
+
+
+var ws = new WebSocket(ws_scheme + location.host + "/ws"); // Open the websocket connection
```

Comment:
@shippy Could this be converted into a ternary expression? Unless the scheme needs to be used elsewhere in the frontend code? So something like this:

```
// Open the websocket connection
var ws = new WebSocket((window.location.protocol == "https:" ? "wss://" : "ws://") + location.host + "/ws"); 
```